### PR TITLE
Cname support + DNS config generation refactoring

### PR DESCRIPTION
**Pull request by alena1108**

### CODE REVIEW

**Comment by alena1108** [comment]
This fix requires latest rancher-dns code.

@vincent99

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/1266

**Comment by alena1108** [comment]
@ibuildthecloud tested against latest Rancher-DNS, both A record and cname resolution works fine. Please review

File: resources/content/db/changelog.xml [hunk 1]
```diff
@@ -58,4 +58,5 @@
     <include file="db/core-051.xml"/>
     <include file="db/core-052.xml"/>
     <include file="db/core-053.xml"/>
+    <include file="db/cname.xml"/>
```
Comment (ibuildthecloud) [inline resources/content/db/changelog.xml]: RENAME!!!!!!

File: resources/content/schema/base/externalService.json [hunk 2]
```diff
@@ -2,7 +2,15 @@
...
         "externalIpAddresses": {
             "type": "array[string]",
-            "required": true,
+            "required": false,
+            "nullable": false,
+            "attributes" : {
+                "scheduleUpdate" : true
+            }
+        },
+        "hostName": {
```
Comment (ibuildthecloud) [inline resources/content/schema/base/externalService.json]: can this be hostname

File: resources/content/db/changelog.xml (hunk 1 above)
Comment (alena1108) [inline resources/content/db/changelog.xml]: thats the first thing I do after the rebase

**Pull request merged**